run/events-storage/anthos.md

```diff
@@ -0,0 +1,96 @@
+# Events for Cloud Run on Anthos – GCS tutorial
+
+This sample shows how to create a service that processes GCS events.
+
+## Setup
+
+Login to gcloud:
+
+```sh
+gcloud auth login
+```
+
+Configure project id:
+
+```sh
+gcloud config set project [PROJECT-ID]
+```
+
+Configure environment variables:
+
+```sh
+MY_RUN_SERVICE=gcs-service
+MY_RUN_CONTAINER=gcs-container
+MY_GCS_TRIGGER=gcs-trigger
+MY_GCS_BUCKET=gcs-bucket-$(gcloud config get-value project)
+```
+
+## Quickstart
+
+Set cluster name, location and platform:
+
+```sh
+gcloud config set run/cluster events-cluster
+gcloud config set run/cluster_location europe-west1-b
+gcloud config set run/platform gke
+```
+
+Deploy your Cloud Run service:
+
+```sh
+gcloud builds submit \
+ --tag gcr.io/$(gcloud config get-value project)/$MY_RUN_CONTAINER
+gcloud run deploy $MY_RUN_SERVICE \
+ --image gcr.io/$(gcloud config get-value project)/$MY_RUN_CONTAINER
+```
+
+Create a bucket:
+
+```sh
+gsutil mb -p $(gcloud config get-value project) -l \
+us-central1 gs://"$MY_GCS_BUCKET"
+```
+
+Before creating a trigger, you need to give the default service account for
+Cloud Storage permission to publish to Pub/Sub.
+
+Find the Service Account that Cloud Storage uses to publish
+to Pub/Sub. You can use the steps outlined in [Cloud Console or the JSON
+API](https://cloud.google.com/storage/docs/getting-service-account). Assume the
+service account you found from above was
+`service-XYZ@gs-project-accounts.iam.gserviceaccount.com`, set this to an
+environment variable:
+
+```sh
+export GCS_SERVICE_ACCOUNT=service-XYZ@gs-project-accounts.iam.gserviceaccount.com
+gcloud projects add-iam-policy-binding $(gcloud config get-value project) \
+--member=serviceAccount:${GCS_SERVICE_ACCOUNT} \
+--role roles/pubsub.publisher
+```
+
+Create Cloud Storage trigger:
+
+```sh
+gcloud alpha events triggers create $MY_GCS_TRIGGER \
+--target-service $MY_RUN_SERVICE \
+--type=com.google.cloud.storage.object.finalize \
```

Comment:
Ran into this error 

`ERROR: (gcloud.alpha.events.triggers.create) Unknown event type: com.google.cloud.storage.object.finalize. If you're trying to use a custom event type, add the "--custom-type" flag.`

Reply:
FYI, when I run the  `gcloud alpha events types list`, it responds with
`Listed 0 items.`

Reply:
My guess is that you don't have ` run/platform` variable set to `gke`. So, it's trying to deploy to `managed` and `managed` does not support `com.google.cloud.storage.object.finalize` yet.

Reply:
`gcloud config list` shows `platform = gke`

Looks like this person had the same problem: https://github.com/GoogleCloudPlatform/python-docs-samples/issues/4228

Reply:
It's similar but not the same problem. #4228 is with Cloud Run Managed, whereas this PR is with Cloud Run for Anthos. Both features are in alpha and projects need to be whitelisted.

Reply:
Ah! Should that be noted somewhere?  Or are you going wait until it's publicly available to merge this PR?

Reply:
Grant already submitted samples for Managed in preparation for upcoming Beta. I'm merely mirroring it for Anthos. I think we can merge this as is and if we need to note anything before Beta, this can be done holistically for Managed and Anthos.